Libraries/LibWeb/Layout/LayoutState.cpp

```diff
@@ -278,63 +270,55 @@ void LayoutState::commit(Box& root)
+            paintable_box->set_content_size(used_values.content_width(), used_values.content_height());
+            if (used_values.override_borders_data().has_value())
+                paintable_box->set_override_borders_data(used_values.override_borders_data().value());
+            if (used_values.table_cell_coordinates().has_value())
+                paintable_box->set_table_cell_coordinates(used_values.table_cell_coordinates().value());
 
-            paintable_box.set_offset(used_values.offset);
-            paintable_box.set_content_size(used_values.content_width(), used_values.content_height());
-            if (used_values.override_borders_data().has_value()) {
-                paintable_box.set_override_borders_data(used_values.override_borders_data().value());
-            }
-            if (used_values.table_cell_coordinates().has_value()) {
-                paintable_box.set_table_cell_coordinates(used_values.table_cell_coordinates().value());
-            }
-
-            if (is<Painting::PaintableWithLines>(paintable_box)) {
-                auto& paintable_with_lines = static_cast<Painting::PaintableWithLines&>(paintable_box);
+            if (auto* paintable_with_lines = as_if<Painting::PaintableWithLines>(*paintable_box)) {
                 for (size_t line_index = 0; line_index < used_values.line_boxes.size(); ++line_index) {
                     auto& line_box = used_values.line_boxes[line_index];
                     for (auto& fragment : line_box.fragments()) {
-                        if (fragment.layout_node().is_text_node())
-                            text_nodes.set(static_cast<Layout::TextNode*>(const_cast<Layout::Node*>(&fragment.layout_node())));
+                        if (auto const* text_node = as_if<TextNode>(fragment.layout_node()))
+                            text_nodes.set(const_cast<TextNode*>(text_node));
                         auto did_relocate_fragment = try_to_relocate_fragment_in_inline_node(fragment, line_index);
-                        if (!did_relocate_fragment) {
-                            paintable_with_lines.add_fragment(fragment);
-                        }
+                        if (!did_relocate_fragment)
+                            paintable_with_lines->add_fragment(fragment);
                     }
                 }
             }
 
-            if (used_values.computed_svg_transforms().has_value() && is<Painting::SVGGraphicsPaintable>(paintable_box)) {
-                auto& svg_graphics_paintable = static_cast<Painting::SVGGraphicsPaintable&>(paintable_box);
-                svg_graphics_paintable.set_computed_transforms(*used_values.computed_svg_transforms());
-            }
+            if (auto* svg_graphics_paintable = as_if<Painting::SVGGraphicsPaintable>(paintable.ptr());
+                svg_graphics_paintable && used_values.computed_svg_transforms().has_value())
+                svg_graphics_paintable->set_computed_transforms(*used_values.computed_svg_transforms());
```

Comment:
This is a bit of a nitpick, but the indentation makes this hard to read. Can you put braces around this to make it more obvious where the block starts and ends? Alternatively, it could be split it into 2 if statements.